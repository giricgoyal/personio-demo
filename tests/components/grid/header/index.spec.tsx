import { render } from '@testing-library/react'
import React from 'react'
import GridHeader from 'src/components/grid/header'

describe('components/grid/header', () => {
    let container, onSortChangeHandler, onFilterChangeHandler
    const sortableColumns = [
        {
            label: 'col1',
            value: 'col1',
        },
    ]

    const filterOptions = [
        {
            label: 'col1',
            value: 'col1',
            filterValues: ['val1', 'val2'],
        },
    ]

    beforeEach(() => {
        onSortChangeHandler = jest.fn()
        onFilterChangeHandler = jest.fn()

        container = render(
            <GridHeader
                title="Some title"
                dataCount={0}
                totalCount={0}
                sortableColumns={sortableColumns}
                sortBy="col1"
                filterBy={''}
                filterOptions={filterOptions}
                onSortChange={onSortChangeHandler}
                onFilterChange={onFilterChangeHandler}
            />,
        )
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })
})
