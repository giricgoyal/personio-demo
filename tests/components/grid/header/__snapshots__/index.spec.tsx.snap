// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/header should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid__header"
      >
        <div
          class="grid__header__left"
        >
          <div
            class="grid__header__left__title"
          >
            Some title
          </div>
          <div
            class="grid__header__left__data-count"
          >
            (
            0
            /
            0
            )
          </div>
        </div>
        <div
          class="grid__header__right"
        >
          <div
            class="grid__header__right__filter"
          >
            <i
              class="fas fa-filter "
            />
            <label
              class="select"
            >
              <span
                class="select__label"
              />
              <select
                class="select__element"
                data-testid="filterby-select"
              >
                <option
                  value="col1"
                >
                  col1
                </option>
              </select>
            </label>
            <label
              class="combo-box"
            >
              <input
                data-testid="filter-value-input"
                list="filter-value-input"
                value=""
              />
            </label>
          </div>
          <div
            class="grid__header__right__sort"
          >
            <i
              class="fas fa-sort "
            />
            <label
              class="select"
            >
              <span
                class="select__label"
              />
              <select
                class="select__element"
                data-testid="select"
              >
                <option
                  selected=""
                  value="col1"
                >
                  col1
                </option>
              </select>
            </label>
            <button
              class="button button--tertiary"
              data-testid="sort-order-button"
            >
              <i
                class="fas fa-arrow-up "
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid__header"
    >
      <div
        class="grid__header__left"
      >
        <div
          class="grid__header__left__title"
        >
          Some title
        </div>
        <div
          class="grid__header__left__data-count"
        >
          (
          0
          /
          0
          )
        </div>
      </div>
      <div
        class="grid__header__right"
      >
        <div
          class="grid__header__right__filter"
        >
          <i
            class="fas fa-filter "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="filterby-select"
            >
              <option
                value="col1"
              >
                col1
              </option>
            </select>
          </label>
          <label
            class="combo-box"
          >
            <input
              data-testid="filter-value-input"
              list="filter-value-input"
              value=""
            />
          </label>
        </div>
        <div
          class="grid__header__right__sort"
        >
          <i
            class="fas fa-sort "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="select"
            >
              <option
                selected=""
                value="col1"
              >
                col1
              </option>
            </select>
          </label>
          <button
            class="button button--tertiary"
            data-testid="sort-order-button"
          >
            <i
              class="fas fa-arrow-up "
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
